### ISSUE TITLE
Add gramma checker to dart analyzer.

### ISSUE THREAD

**Issue by polina-c** [dart-lang/sdk#42538]
It would be very convenient if dart analyzer was as smart as google docs spell/gramma checker.

**Comment by srawlins** [comment]
Can you elaborate what you would hope such analysis would report? The analyzer does check that programs adhere to the _Dart_ grammar. A spell checker would not be useful for identifier names as it is extremely common for identifiers to not be English words, and to contain non-words.

**Comment by polina-c** [comment]
I mean documentation comments. That start with ‘///‘. I am non English speaker and a lot of comments in me code reviews are about missed articles (‘a’ and ‘the’) and missed ‘s’ at the end of verbs.

This creates extra effort for my reviewers and me.

Google docs handles such kind if mistakes.

As there is a styling rule, that doc comments should be sentences, it is a valid point that gramma could be auto-validated.

https://dart.dev/guides/language/effective-dart/documentation#do-format-comments-like-sentences

Does this help?

**Comment by srawlins** [comment]
Thanks, @polina-c I understand. I know that this is a bit built-in to some IDEs. I get spelling suggestions in IntelliJ.

@jwren @bwilkerson does this make sense to do in Analysis Server / linter, or do most IDEs provide spell-check? What about grammar checking?

**Comment by bwilkerson** [comment]
It is reasonable to want this kind of feature, but
- as you noted most IDEs already have this support,
- it isn't specific to Dart, and
- it's a very big project for us to take on unless there's some technology that we could integrate to support it.

It isn't my call, but I'd be very surprised if we were to decide to take on a project of this side.

**Comment by polina-c** [comment]
Yes, it seems Google natural language API does not offer gramma checks: https://cloud.google.com/natural-language.

**Comment by srawlins** [comment]
We won't be implementing this as a warning or a lint rule in the Dart analyzer. We now have a high bar for implementing and maintaining built-in static analysis rules. As of Dart 3.10 (Flutter 3.38), a [new analyzer plugins system](https://dart.dev/tools/analyzer-plugins) is available, and the requested static analysis here could be implemented as an analyzer plugin.
